### PR TITLE
ARTEMIS-1706 - Add support for wantClientAuth

### PR DESCRIPTION
**Pull request by cshannon**
Support setting wantClientAuth on a netty acceptor

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyAcceptor.java [hunk 1]
```diff
@@ -468,8 +473,12 @@ public synchronized SslHandler getSslHandler(ByteBufAllocator alloc) throws Exce
 
       engine.setUseClientMode(false);
 
-      if (needClientAuth)
+      if (needClientAuth) {
          engine.setNeedClientAuth(true);
+      }
+      if (wantClientAuth) {
```
Comment (tabish121) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyAcceptor.java]: I would probably do this in an else as the setting would override the need option and if someone did something silly like specify both I'd generally prefer it chose the stronger of the two and setting want after need overrides it.

File: docs/user-manual/en/configuring-transports.md [hunk 2]
```diff
@@ -403,6 +403,15 @@ following additional properties:
...
     are `true` or `false`. Default is `false`.
+    
+-   `wantClientAuth`
+
+    This property is only for an `acceptor`. It tells a client
+    connecting to this acceptor that 2-way SSL is requested but not required. 
+    Valid values are `true` or `false`. Default is `false`. 
+    
+    Note that this setting will override `needClientAuth` if both properties 
+    are set to true.
```
Comment (tabish121) [inline docs/user-manual/en/configuring-transports.md]: I failed to read this before my previous comment but as before I'd have opted to go the other way on the choice and defaulted to need if both are there.  For reference the 5.x code makes that choice as well when both are present on the URI

**Comment by clebertsuconic** [comment]
can you rebase against master? I have sent a fix to the JORAM tests.. this would be a helpful as a test as well.

**Comment by clebertsuconic** [comment]
@cshannon rebase it... are you addressing @tabish121's comments or that doesn't apply?

**Comment by cshannon** [comment]
I am going to update my PR with Tim’s suggestion to have it match 5.x and rebase in the morning

**Comment by cshannon** [comment]
@clebertsuconic and @tabish121 - PR has been updated and rebased.

**Comment by tabish121** [comment]
Looks good.  +1

**Pull request merged**